control/velocity_controller_lqr/velocity_controller_lqr/velocity_controller_lqr_lib.py

```diff
@@ -0,0 +1,282 @@
+#!/usr/bin/env python3
+import numpy as np
+
+import control as ct
+
+
+class LQRController:
+    def __init__(self, parameters, inertia_matrix):
+        self.integral_error_surge = 0.0
+        self.integral_error_pitch = 0.0
+        self.integral_error_yaw = 0.0
+
+        self.surge_windup = False  # Windup variable
+        self.pitch_windup = False  # Windup variable
+        self.yaw_windup = False  # Windup variable
+
+        self.q_surge = parameters[0]  # Surge cost
+        self.q_pitch = parameters[1]  # Pitch cost
+        self.q_yaw = parameters[2]  # Yaw cost
+
+        self.r_surge = parameters[3]  # Surge control cost
+        self.r_pitch = parameters[4]  # Pitch control cost
+        self.r_yaw = parameters[5]  # Yaw control cost
+
+        self.i_surge = parameters[6]  # Integral gain for surge
+        self.i_pitch = parameters[7]  # Integral gain for pitch
+        self.i_yaw = parameters[8]  # Integral gain for yaw
+
+        self.i_weight = parameters[9]  # Weight for integral gain
+
+        self.max_force = parameters[10]  # Maximum force that can be applied
+
+        self.inertia_matrix_inv = inertia_matrix  # Inverse of the inertia matrix
+
+        # VARIABLES
+        self.inertia_matrix_inv = np.linalg.inv(
+            inertia_matrix
+        )  # Inverse of the inertia matrix
+
+        self.state_weight_matrix = np.block(
+            [
+                [np.diag([self.q_surge, self.q_pitch, self.q_yaw]), np.zeros((3, 3))],
+                [
+                    np.zeros((3, 3)),
+                    np.diag([self.i_weight, self.i_weight, self.i_weight]),
+                ],
+            ]
+        )  # Augmented state cost matrix
+        self.input_weight_matrix = np.diag(
+            [self.r_surge, self.r_pitch, self.r_yaw]
+        )  # control cost matrix
```

Comment:
I like to have specific method that sets for example the intertia matrix, and parameters. That way you can initialize the controller in the constructor of your ros node, and set the values in their respected methods. It is also not very clear what the parameters here represent or where they come from, so if hypothetically someone wanted to use your controller class they would have a hard time :sweat_smile: 

Also as in the ros node, consider having a method that initialized these values, so that the init method is not so big